app/lib/shared/user_merger.dart

```diff
@@ -0,0 +1,191 @@
+// Copyright (c) 2019, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:gcloud/db.dart';
+import 'package:pool/pool.dart';
+
+import '../account/models.dart';
+import '../package/models.dart';
+import '../publisher/models.dart';
+
+/// Utility class to merge user data.
+class UserMerger {
+  final DatastoreDB _db;
+  final int _concurrency;
+  final bool _deleteUsers;
+
+  UserMerger({DatastoreDB db, int concurrency = 1, bool deleteUsers})
+      : _db = db,
+        _concurrency = concurrency,
+        _deleteUsers = deleteUsers ?? false;
+
+  /// Fixes all OAuthUserID issues.
+  Future fixAll() async {
+    final ids = await scanOauthUserIdsWithProblems();
+    for (String id in ids) {
+      await fixOAuthUserID(id);
+    }
+  }
+
+  /// Returns the OAuth userIds that have more than one User.
+  Future<List<String>> scanOauthUserIdsWithProblems() async {
+    print('Scanning Users...');
+    final query = _db.query<User>();
+    final counts = <String, int>{};
+    await for (final user in query.run()) {
+      if (user.oauthUserId == null) continue;
+      counts[user.oauthUserId] ??= 0;
+      counts[user.oauthUserId]++;
+    }
+    final result = counts.keys.where((k) => counts[k] > 1).toList();
+    print('$result OAuthUserID with more than one User.');
+    return result;
```

Comment:
I think a better remedy is to iterate through all users.. and for each user:
 1. If `user.oauthUserId == null` then continue,
 2. Load the `OAuthUserID` entity and check if it points back to `user.user_id`, if not, then we have a problem.

Then we can make this a stream of userIds that has issues. Hence, we don't need to load all users at once.

Reply:
We will need to iterate through all users anyway. We either lookup and check the mapping or keep a count. Once we found a problem, we'll need to load all the related User entities to check the e-mail match. I don't see much limit in the current approach, let's keep it for now.

Reply:
Okay, I don't see a huge problem... It just seems like we could make a more robust solution the other way, but I'm fine this is hopefully a one-off :)